sway-lsp/src/capabilities/document_symbol.rs

```diff
@@ -1,60 +1,499 @@
-use crate::core::token::{SymbolKind, Token, TokenIdent};
-use dashmap::mapref::multiple::RefMulti;
-use lsp_types::{self, Location, SymbolInformation, Url};
-
-pub fn to_symbol_information<'a, I>(tokens: I, url: &Url) -> Vec<SymbolInformation>
-where
-    I: Iterator<Item = RefMulti<'a, TokenIdent, Token>>,
-{
-    tokens
-        .map(|entry| {
-            let (ident, token) = entry.pair();
-            symbol_info(ident, token, url.clone())
+use crate::core::{token::get_range_from_span, token_map::TokenMap};
+use lsp_types::{self, DocumentSymbol, Url};
+use std::path::PathBuf;
+use sway_core::{
+    language::ty::{
+        TyAbiDecl, TyAstNodeContent, TyConstantDecl, TyDecl, TyEnumDecl, TyFunctionDecl,
+        TyFunctionParameter, TyProgram, TyStorageDecl, TyStructDecl, TyTraitInterfaceItem,
+        TyTraitItem, TyTraitType,
+    },
+    Engines, TypeArgument,
+};
+use sway_types::{Span, Spanned};
+
+/// Generates a hierarchical document symbol tree for LSP code outline/navigation.
+/// Processes declarations (functions, structs, enums, etc.) into nested symbols,
+/// preserving parent-child relationships like functions with their variables,
+/// structs with their fields, and traits with their methods.
+pub fn to_document_symbols(
+    uri: &Url,
+    path: &PathBuf,
+    ty_program: &TyProgram,
+    engines: &Engines,
+    token_map: &TokenMap,
+) -> Vec<DocumentSymbol> {
+    let source_id = engines.se().get_source_id(path);
+
+    // Find if there is a configurable symbol in the token map that belongs to the current file
+    // We will add children symbols to this when we encounter a configurable declarations below.
```

Comment:
- encounter a configurable
+ encounter configurable

Reply:
thanks, done in #[2d682a9](https://github.com/FuelLabs/sway/pull/6678/commits/2d682a91ecc51d86e098add9b81d23a0c27f823b)